raphtory/src/python/graph/node_state/node_state_result_option_datetime.rs

```diff
@@ -0,0 +1,435 @@
+use crate::{
+    db::{
+        api::{
+            state::{NodeGroups, NodeState},
+            view::DynamicGraph,
+        },
+        graph::{node::NodeView, nodes::Nodes},
+    },
+    prelude::{GraphViewOps, NodeStateOps, NodeViewOps},
+    python::{
+        types::{repr::Repr, wrappers::iterators::PyBorrowingIterator},
+        utils::PyNodeRef,
+    },
+};
+use chrono::{DateTime, Utc};
+use pyo3::{
+    exceptions::{PyKeyError, PyTypeError},
+    prelude::*,
+    types::{PyDict, PyNotImplemented},
+    IntoPyObjectExt,
+};
+use raphtory_api::core::storage::timeindex::TimeError;
+use raphtory_core::entities::nodes::node_ref::{AsNodeRef, NodeRef};
+use rayon::{iter::ParallelIterator, prelude::ParallelSliceMut};
+use std::{cmp::Ordering, collections::HashMap};
+// $name = NodeStateResultOptionDateTime
+// $value = Result<Option<DateTime<Utc>>, TimeError>
+// $to_owned = |v| v.clone()
+// $inner_t = NodeState<'static, Result<Option<DateTime<Utc>>, TimeError>, DynamicGraph, DynamicGraph>
+// $py_value = Optional[datetime]
+// $computed = NodeStateResultOptionDateTime
+
+#[pyclass(module = "raphtory.node_state", frozen)]
+pub struct NodeStateResultOptionDateTime {
```

Comment:
this probably shouldn't exist, we should pull the result out on construction instead